### PR TITLE
Time the app's display after the data is retrieved from the server

### PR DESCRIPTION
**Pull request by AnirudhPudari**
Issue Link https://github.com/anandtharad/QuizApp/issues/4

**Addressed Issues**
- Sync issue of first question is addressed
- Added a progress before all the questions are loaded and rendered for a better user experience.

**Video after Fix**

https://user-images.githubusercontent.com/61533235/230886446-c5d1c8da-3d1a-4d28-ac75-a7d5cb69d72b.mov

### CODE REVIEW

File: app/src/main/java/com/example/quiz/MainActivity.java [hunk 1]
```diff
@@ -42,55 +42,52 @@ public class MainActivity extends AppCompatActivity {
     private int userScore;
     private TextView totalQuestions;
     final int USER_PROGRESS = (int) Math.ceil(100.0 / (double) 10); //increments progress bar by USER_PROGRESS each time answered
-    private int mQuestionIndex;
+    private int mQuestionIndex = 0;
```
Comment (AnirudhPudari) [inline app/src/main/java/com/example/quiz/MainActivity.java:45]: initial state index set to 0

File: app/src/main/java/com/example/quiz/MainActivity.java [hunk 2]
```diff
@@ -136,6 +131,28 @@ public void onClick(View view) {
 
     }
 
+    private void initViews() {
```
Comment (AnirudhPudari) [inline app/src/main/java/com/example/quiz/MainActivity.java:134]: Entire view initializing findViewById calls moved to a common method for easy access.

File: app/src/main/java/com/example/quiz/MainActivity.java [hunk 3]
```diff
@@ -42,55 +42,52 @@ public class MainActivity extends AppCompatActivity {
     private int userScore;
     private TextView totalQuestions;
     final int USER_PROGRESS = (int) Math.ceil(100.0 / (double) 10); //increments progress bar by USER_PROGRESS each time answered
-    private int mQuestionIndex;
+    private int mQuestionIndex = 0;
     private String mQuizQuestion;
     private List<QuizQuestion> questionCollection;
 
+    private View loadingView;
```
Comment (AnirudhPudari) [inline app/src/main/java/com/example/quiz/MainActivity.java:49]: Loading view added with new layout to indicate that the questions are being loaded for a better user experience.

**Comment by anandtharad** [comment]
Thanks man, I was looking for the solution for so long.

**Pull request merged**